### ISSUE TITLE
Support LogConfiguration in container properties

### ISSUE THREAD

**Issue by MrMarkW** [nextflow-io/nextflow#2801]
## New feature

Add support for log configuration for docker and AWS Batch

## Usage scenario 

When AWS Batch job definitions are generated, custom log configuration may be required by a consumer. 

For example, we want to force the awslogs driver and send the results to a specific region and log group.

## Suggest implementation 

Option 1:  The configuration can be set via process.containerOptions
Update `AwsContainerOptionsMapper` to support parsing log-driver and log-opt configuration.

`containerOptions = '--log-driver awslogs --log-opt awslogs-region=us-east-1 awslogs-group=/aws/batch/my-batch-env-name'`

[AwsBatchHelper](https://github.com/nextflow-io/nextflow/blob/5485b699465b83b4ee7584174290d365487eb14d/plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchHelper.groovy#L195-L213) will need to change also. 

Option 2: Add explicit LogConfiguration support as AWS Batch options and in AwsBatchHelper.

```
aws {
    batch {
      logConfiguration {
        logDriver = awslogs
        options {
          awslogs-group = /aws/batch/queue-name
        }
    }
}
```

Option 3: Automatically set AWS batch job definitions to use awslogs and logs-groups

The log group could be automatically named based on the queue name. Example: `/aws/batch/${queue}`

This could be enabled with a configuration flag.

**Comment by pditommaso** [comment]
Think [this comment](https://github.com/nextflow-io/nextflow/pull/2802) sheds some light on what's the goal of this issue. 

Do you want to be able to set a custom path for logs in cloudwatch ?

**Comment by pditommaso** [comment]
It should not to be to hard to add a config setting in the nextflow config. But need to check if it will break the logs fetching in Tower system

**Comment by varontron** [comment]
I second the need for custom log group names/paths.  We use cloudwatch, and batch (and many other aws services) extensively in my organization, and nextflow is only one such client of these services.  Furthermore, we use datadog for reporting and monitoring, and filtering forwarded logs by log group name is a lynchpin for that operation.

**Comment by bentsherman** [comment]
Fixed by #3092
